### PR TITLE
Install command

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

Adds the `flasher install` command which allows flasher to be run as a one-time firmware install command.
For now this is limited to installing firmware for a single component, going ahead it'll be expanded to accept a list of servers and firmware to be installed.

### CODE REVIEW

File: internal/outofband/action_handlers.go [hunk 1]
```diff
@@ -249,6 +249,16 @@ func (h *actionHandler) downloadFirmware(a sw.StateSwitch, c sw.TransitionArgs)
 		return err
 	}
 
+	if action.FirmwareTempFile != "" {
```
Comment (DoctorVin) [inline internal/outofband/action_handlers.go:252]: Is this apropos of a service-based fw install as well?

**Review by DoctorVin — COMMENTED** [review]
Some food for thought. Probably fine as is, IMO better with some minor revisions.

File: internal/install/task_handler.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+func (h *taskHandler) Plan(t sw.StateSwitch, args sw.TransitionArgs) error {
+	tctx, ok := args.(*sm.HandlerContext)
+	if !ok {
+		return sm.ErrInvalidtaskHandlerContext
+	}
+
+	task, ok := t.(*model.Task)
+	if !ok {
+		return errors.Wrap(ErrSaveTask, ErrTaskTypeAssertion.Error())
+	}
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/install/task_handler.go:71]: Agreed, this is more of a stop gap to have this CLI command available. The intention is to remove the Task statemachine completely - we'll only keep the Action statemachine, and so I'd prefer to not make any improvements in this area for now.

File: cmd/install.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		Force:     force,
+	}
+
+	installer := install.New(l)
+
+	installer.Install(ctx, p)
+}
+
+func init() {
+	cmdInstall.PersistentFlags().BoolVarP(&dryrun, "dry-run", "", false, "dry run install")
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply cmd/install.go]: fixed

File: internal/install/install.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			Model:       params.Model,
+			Vendor:      params.Vendor,
+		},
+		Logger: le,
+	}
+
+	i.runTaskStatemachine(handler, handlerCtx)
+}
+
+type publisher struct{ logger logrus.Entry }
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/install/install.go]: fixed

File: internal/install/task_handler.go [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+}
+
+// Query looks up the device component inventory and sets it in the task handler context.
+func (h *taskHandler) Query(t sw.StateSwitch, args sw.TransitionArgs) error {
+	tctx, ok := args.(*sm.HandlerContext)
+	if !ok {
+		return sm.ErrInvalidtaskHandlerContext
+	}
+
+	_, ok = t.(*model.Task)
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/install/task_handler.go]: fixed

**Review by DoctorVin — APPROVED** [review]

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**